sw/device/tests/sim_dv/alert_handler_escalation.c

```diff
@@ -0,0 +1,163 @@
+ */
+
+OTTF_DEFINE_TEST_CONFIG();
+
+static dif_clkmgr_t clkmgr;
+static dif_keymgr_t keymgr;
+static dif_rstmgr_t rstmgr;
+static dif_alert_handler_t alert_handler;
+static dif_rv_core_ibex_t rv_core_ibex;
+static dif_uart_t uart;
+
+typedef struct node {
+  const char *name;
+  dif_alert_handler_alert_t alert;
+  dif_alert_handler_class_t class;
+} node_t;
+
+static volatile uint32_t global_alert_called;
+static const dif_alert_handler_escalation_phase_t kEscProfiles[] = {
+    {.phase = kDifAlertHandlerClassStatePhase0,
+     .signal = 0,
+     .duration_cycles = 5000},
+    // TODO:
+    // this second duration must be long enough to
+    // accommodate a jtag transaction
+    // how can this be done in a non-hardcoded way?
+    {.phase = kDifAlertHandlerClassStatePhase1,
+     .signal = 1,
+     .duration_cycles = 10000},
+    {.phase = kDifAlertHandlerClassStatePhase2,
+     .signal = 3,
+     .duration_cycles = 3000}};
+
+static const dif_alert_handler_class_config_t kConfigProfiles[] = {{
+    .auto_lock_accumulation_counter = kDifToggleDisabled,
+    .accumulator_threshold = 0,
+    .irq_deadline_cycles = 0,
+    .escalation_phases = kEscProfiles,
+    .escalation_phases_len = 3,
+    .crashdump_escalation_phase = kDifAlertHandlerClassStatePhase1,
+}};
+
+/**
+ * External ISR.
+ *
+ * Handles all peripheral interrupts on Ibex. PLIC asserts an external interrupt
+ * line to the CPU, which results in a call to this OTTF ISR. This ISR
+ * overrides the default OTTF implementation.
+ */
+void ottf_external_nmi_handler(void) {
+  // DO NOT REMOVE, DV sync message
+  LOG_INFO("You are experiencing an NMI");
+
+  // Now intentionally hang the device
+  CHECK_DIF_OK(dif_clkmgr_gateable_clock_set_enabled(
+      &clkmgr, kTopEarlgreyGateableClocksIoDiv4Peri, kDifToggleDisabled));
+
+  // access uart after clocks have been disabled
+  CHECK_DIF_OK(dif_uart_disable_rx_timeout(&uart));
+  LOG_FATAL("This message should never be seen");
+}
+
+bool test_main(void) {
+  CHECK_DIF_OK(dif_clkmgr_init(
+      mmio_region_from_addr(TOP_EARLGREY_CLKMGR_AON_BASE_ADDR), &clkmgr));
+
+  CHECK_DIF_OK(dif_rstmgr_init(
+      mmio_region_from_addr(TOP_EARLGREY_RSTMGR_AON_BASE_ADDR), &rstmgr));
+
+  CHECK_DIF_OK(dif_alert_handler_init(
+      mmio_region_from_addr(TOP_EARLGREY_ALERT_HANDLER_BASE_ADDR),
+      &alert_handler));
+
+  CHECK_DIF_OK(dif_rv_core_ibex_init(
+      mmio_region_from_addr(TOP_EARLGREY_RV_CORE_IBEX_CFG_BASE_ADDR),
+      &rv_core_ibex));
+
+  CHECK_DIF_OK(dif_uart_init(
+      mmio_region_from_addr(TOP_EARLGREY_UART0_BASE_ADDR), &uart));
+
+  CHECK_DIF_OK(dif_keymgr_init(
+      mmio_region_from_addr(TOP_EARLGREY_KEYMGR_BASE_ADDR), &keymgr));
+
+  // Check if there was a HW reset caused by the escalation.
+  dif_rstmgr_reset_info_bitfield_t rst_info;
+  CHECK_DIF_OK(dif_rstmgr_reset_info_get(&rstmgr, &rst_info));
+  CHECK_DIF_OK(dif_rstmgr_reset_info_clear(&rstmgr));
+
+  if (rst_info & kDifRstmgrResetInfoPor) {
```

Comment:
just dropping a note here for @drewmacrae @a-will for a future cleanup: if we move the reset info to ret SRAM and clear it in the test ROM as proposed in #13096, then this will also need to be updated.